trieste/acquisition/optimizer.py

```diff
@@ -246,8 +249,26 @@ def optimize_continuous(
         if V < 0:
             raise ValueError(f"vectorization must be positive, got {V}")
 
-        candidates = space.sample(num_initial_samples)[:, None, :]  # [num_initial_samples, 1, D]
-        tiled_candidates = tf.tile(candidates, [1, V, 1])  # [num_initial_samples, V, D]
+        candidates = space.sample(num_initial_samples)
+        if tf.rank(candidates) == 3:
+            # If samples is a tensor of rank 3, then it is a batch of samples. In this case
+            # the length of the second dimension must be equal to the vectorization of the target
+            # function.
+            tf.debugging.assert_equal(
+                tf.shape(candidates)[1],
+                V,
+                message=(
+                    f"""
+                    The batch shape of initial samples {tf.shape(candidates)[1]} must be equal to
+                    the vectorization of the target function {V}.
+                    """
+                ),
+            )
+            tiled_candidates = candidates  # [num_initial_samples, V, D]
+        else:
+            tiled_candidates = tf.tile(
```

Comment:
maybe assert that the rank is 2 here? (partly for robustness in case we decide to break the shape sample api again, but mainly for code readability)